sources/packages/api/src/route-controllers/application/application.students.controller.ts

```diff
@@ -448,4 +456,120 @@ export class ApplicationStudentsController extends BaseController {
       applicationNumber: application.applicationNumber,
     };
   }
+
+  /**
+   * Get in progress details of an application by application id.
+   * @param applicationId application id.
+   * @returns application details.
+   */
+  @Get(":id/in-progress")
+  @ApiNotFoundResponse({
+    description: "Application id not found.",
+  })
+  async getInProgressApplicationDetails(
+    @Param("id", ParseIntPipe) applicationId: number,
+  ): Promise<InProgressApplicationDetailsAPIOutDTO> {
+    const application = await this.applicationService.getApplicationByIdAndUser(
+      applicationId,
+    );
+    if (!application) {
+      throw new NotFoundException(
+        `Application id ${applicationId} was not found.`,
+      );
+    }
+
+    const incomeVerificationDetails =
+      await this.craIncomeVerificationService.allIncomeVerificationsForAnApplication(
+        applicationId,
+      );
+    const incomeVerification =
+      this.applicationControllerService.processApplicationIncomeVerificationDetails(
+        incomeVerificationDetails,
+      );
+
+    const supportingUserDetails =
+      await this.supportingUserService.getSupportingUsersByApplicationId(
+        applicationId,
+      );
+
+    const supportingUser =
+      this.applicationControllerService.processApplicationSupportingUserDetails(
+        supportingUserDetails,
+      );
+
+    return {
+      id: application.id,
+      applicationStatus: application.applicationStatus,
+      pirStatus: application.pirStatus,
+      PIRDeniedReason: getPIRDeniedReason(application),
+      offeringStatus: application.currentAssessment?.offering.offeringStatus,
+      exceptionStatus: application.applicationException?.exceptionStatus,
+      ...incomeVerification,
+      ...supportingUser,
+    };
+  }
+
+  /**
+   * Get cancelled details of an application by application id.
+   * @param applicationId application id.
+   * @returns application details.
+   */
+  @Get(":id/cancelled")
+  @ApiNotFoundResponse({
+    description: "Application id not found.",
+  })
+  async getCancelledApplicationDetails(
```

Comment:
Do we need a new method to get just this info?

Reply:
Wanted to ask this. Currently we do not have a get API for applications on student side which returns a summary of application details(excluding the json data in it). It would be nice to build one isn't it?

Reply:
@dheepak-aot I would be ok with creating one like what you said or maybe just providing a query string to the existing one to avoid retrieving all the dynamic data. Either way, I would just avoid creating too specific a method unless there is a performance constraint.

Reply:
as @andrewsignori-aot  said we have `getByApplicationId` for students. I was planning to use it first. then I noticed it is returning a lot of details including COE details, which are not needed for a canceled application. 
I will add this `application.applicationStatusUpdatedOn`  to `getApplicationStatusDetails`, which is consumed by the parent of the component and let the parent pass this details to the child